torch/csrc/jit/tracer.h

```diff
@@ -1,10 +1,10 @@
 #pragma once
 
 #include <ATen/Backtrace.h>
+#include <c10/util/Exception.h>
```

Comment:
I guess my commit messed up the formatting, sorry! Could you please commit the formatting changes separately though?

Reply:
So far there are actually three call sites to `toOptionalTensor`, in `register_prim_ops`, `register_aten_ops`, and `graph_executor`, that's why I want to put it into a single place in jit/ivalue.h, otherwise we will need to put the function repeatedly into three files which is kinda not good. In regards with the formatting changes, it is a pre-commit hook to run the clang-format automatically everytime we modify a file, I believe the most part of formatting changes is in `register_prim_ops`, I can do a separate formatting PR for it if you prefer.

Reply:
Do we expect to get rid of the undefined tensors in TH in any near future? If not, I'd just put `toOptionalTensor` to `ir.h` and put it to `c10` when we move ir.h to c10 (if we do this). As for the formatting - either way works for me :)

Reply:
I believe the call in register_prim_ops is incorrect and does not need to exist. The call in graph_executor should be explicitly written because it is doing something deeply surprising that should not be hidden behind a function call. Then toOptionalTensor can just be defined locally in register_aten_ops.